rand_core/src/lib.rs

```diff
@@ -175,32 +175,33 @@ where
     }
 }
 
-/// A marker trait used to indicate that an [`RngCore`] implementation is
-/// supposed to be cryptographically secure.
+/// A marker trait over [`RngCore`] for securely unpredictable RNGs
 ///
-/// *Cryptographically secure generators*, also known as *CSPRNGs*, should
-/// satisfy an additional properties over other generators: given the first
-/// *k* bits of an algorithm's output
+/// This marker trait indicates that the implementing generator is intended,
+/// when correctly seeded and protected from side-channel attacks such as a
+/// leaking of state, to be a cryptographically secure generator. This trait is
```

Comment:
It may be worth to add:
>leaking of state (i.e. the generator may not be "backtracking resistant")

Reply:
I updated the last paragraph, which already concerned this